entrypoint.sh

```diff
@@ -1,32 +1,58 @@
 #!/bin/sh
 
 TRACEE_EBPF_EXE=${TRACEE_EBPF_EXE:="/tracee/tracee-ebpf"}
-TRACEE_WEBHOOK_CONFIG=${TRACEE_WEBHOOK_CONFIG:="/tmp/tracee/integrations-config.yaml"}
-TRACEE_WEBHOOK_EXE=${TRACEE_WEBHOOK_EXE:="/tracee/falcosidekick"}
 TRACEE_RULES_EXE=${TRACEE_RULES_EXE:="/tracee/tracee-rules"}
 
-if [ "$1" = "trace" ]; then
-	shift
-	$TRACEE_EBPF_EXE $@
-	exit
-fi
+die() {
+	echo $1 >&2
+	exit "$2"
+}
 
-if [ "$1" = "--list" ]; then
-	shift
-	$TRACEE_RULES_EXE --list
-	exit
-fi
+# parse_args parses the arguments passed to the entrypoint
+# it first looks for subcommands, which will be executed immidiately
+# then it parses flags for the default command in format '--flagname flagvalue'
+# parsed flags are accessible under arg_flagname variables
+# positional args are not supported
+parse_args() {
+	while test $# -gt 0; do
+		case "$1" in
+			trace)
+				shift
+				$TRACEE_EBPF_EXE $@
+				exit "$?"
+				;;
+			--list)
+				shift
+				$TRACEE_RULES_EXE --list
+				exit "$?"
+				;;
+			--rules)
+				test $# -lt 2 && die "missing value for argument '$1'." 1
+				arg_rules="$2"
+				shift
+				;;
+			--webhook-url)
+				test $# -lt 2 && die "missing value for argument '$1'." 1
+				arg_webhook_url="$2"
+				shift
+				;;
+			--webhook-content-type)
+				test $# -lt 2 && die "missing value for argument '$1'." 1
+				arg_webhook_content_type="$2"
+				shift
+				;;
+			*)
```

Comment:
What if both --rules and --webhook-url are given? using **-lt 2** will not work in that case.
As all of these arguments are handled by tracee-rules (only tracee-ebpf is not) - maybe we should remove this entrypoint and just call tracee-rules, and if **trace** is given - call tracee-ebpf

Reply:
> What if both --rules and --webhook-url are given? using -lt 2 will not work in that case.

why would it not work? 2<4. not sure what's the issue you are seeing?

Reply:
> remove this entrypoint and just call tracee-rules

who will orchestrate passing events from tracee-ebpf to tracee-rules (the pipe)? the entrypoint is needed for that

> and if trace is given - call tracee-ebpf

where is this `if` going to be if not in entrypoint?

Reply:
> why would it not work? 2<4. not sure what's the issue you are seeing?

It will work, but miss the point of argument verification, for example, in the follwing case:
`tracee --rules --webhook-url http://bla`
There is a missing value for **rules** in that case.

> who will orchestrate passing events from tracee-ebpf to tracee-rules (the pipe)? the entrypoint is needed for that

tracee-rules can - by creating a pipe, and calling fork-execve. It can also create a pipe which is not stdin based, allowing us to add more sources in the future.

> where is this if going to be if not in entrypoint?

tracee-rules can have a new flag called "trace". If given - execve tracee-ebpf (without forking first)

Reply:
ok so you're suggesting to make tracee-rules the entrypoint.. this will turn tracee-rules to be not just a rule engine, but the frontend for tracee. we discussed it before and IIRC you didn't like that idea. anyway, I think it's beyond the scope of this PR

Reply:
> but miss the point of argument verification

I'm wasn't trying to do argument verification, just defensive programming. This also wasn't handled before this PR I think. But I'll change it to verify the arg

Reply:
I think that argument verification should be left to tracee-rules - no need to do this here as well.

> ok so you're suggesting to make tracee-rules the entrypoint.. this will turn tracee-rules to be not just a rule engine, but the frontend for tracee. we discussed it before and IIRC you didn't like that idea. anyway, I think it's beyond the scope of this PR

When we discussed it in the past, I suggested to have 'trace' flag as well as 'rules' flag. In the end, only 'trace' flag was implemented, and tracee-rules became the de-facto frontend for tracee.  I think that we should first answer the question - what is the role of the entrypoint? (especially now, after we removed mounting debugfs). 
I don't think that the role of the entrypoint should be argument verification - this should be done either by tracee-rules or tracee-ebpf.
This leaves us with the role of creating a pipe between tracee-rules and whatever source we will give it in the future. But this role can be implemented by tracee-rules, isn't it?

Reply:
For this PR, I suggest only removing the falcosidekick part from the entrypoint

Reply:
> for this PR, I suggest only removing the falcosidekick part from the entrypoint

part of the motivation to remove it from the container is that we're now able to call it or others using the new webhook flags. If we remove falcosidekick from the container we should also add support in those flags, which is what I did

Reply:
> what is the role of the entrypoint?

since we chose to build tracee-ebpf and tracee-rules as independent projects, the entrypoint is what ties everything together. the fact that it's now simpler is good, we should be happy about simplifying it.
if we move that functionality into tracee-rules, it is not longer an independent rules engine, we might as well rename it to "tracee" like originally suggested. 

as a suggestion - since currently the entrypoint's ux is almost the same as tracee-rules, I could just pass all flags to tracee-rules. but I'm not sure that going forward that'll be the best solution

Reply:
I pushed a new versions that you might like better, where tracee-rules handles most of the args directly